dig_test.go

```diff
@@ -427,6 +427,31 @@ func TestEndToEndSuccess(t *testing.T) {
 		}))
 	})
 
+	t.Run("out type inserts multiple objects into the graph", func(t *testing.T) {
+		type A struct {
+			name string
+		}
+		type B struct {
+			name string
+		}
+		type Ret struct {
+			Out
+
+			foo string // private field to be ignored
+			A          // value type A
+			*B         // pointer type *B
+		}
+
+		c := New()
+		require.NoError(t, c.Provide(func() Ret {
+			return Ret{A: A{"string A"}, B: &B{"string B"}}
+		}), "provide for the Ret struct should succeed")
+		require.NoError(t, c.Invoke(func(a A, b *B) {
+			assert.Equal(t, a.name, "string A", "value type should work for dig.Out")
+			assert.Equal(t, b.name, "string B", "pointer sohuld work for dig.Out")
```

Comment:
spelling: "should"

Also, can we be nit-picky and also validate that we're getting the same ref returned from the constructors?